### PR TITLE
Change the default credits for the RPC link to be more reasonable (1000)

### PR DESCRIPTION
**Pull request by richardpark-msft**
Use a higher link credit count by default for the rpc link (it's defaulting to 1 today)
    
In the future we could control this manually since we know when responses are expected. This should be fine in the meantime.

### CODE REVIEW

File: rpc/rpc.go [hunk 1]
```diff
@@ -148,6 +149,7 @@ func NewLinkWithSession(session *amqp.Session, address string, opts ...LinkOptio
 	receiverOpts := []amqp.LinkOption{
 		amqp.LinkSourceAddress(address),
 		amqp.LinkTargetAddress(link.clientAddress),
+		amqp.LinkCredit(defaultReceiverCredits),
```
Comment (serbrech) [inline rpc/rpc.go:152]: Mmh, that's making it default for anyone constructing the a link.
I don't think that's a good approach. It must be for the RPC calls channel only

Comment (serbrech) [reply rpc/rpc.go:152]: But that's in the RPC package... I see

**Comment by serbrech** [comment]
this is really needed with the unique link for the RPCClient.

File: rpc/rpc.go (hunk 1 above)
Comment (richardpark-msft) [reply rpc/rpc.go:152]: Merging.

File: rpc/rpc.go [hunk 2]
```diff
@@ -40,9 +40,10 @@ import (
 )
 
 const (
-	replyPostfix   = "-reply-to-"
-	statusCodeKey  = "status-code"
-	descriptionKey = "status-description"
+	replyPostfix           = "-reply-to-"
+	statusCodeKey          = "status-code"
+	descriptionKey         = "status-description"
+	defaultReceiverCredits = 1000
```
Comment (jhendrixMSFT) [inline rpc/rpc.go:46]: Is this value similar to what other SDKs use?

Comment (serbrech) [reply rpc/rpc.go:46]: Servicebus management link (server side peer uses 5000

**Review by jhendrixMSFT — APPROVED** [review]

File: rpc/rpc.go (hunk 2 above)
Comment (serbrech) [reply rpc/rpc.go:46]: Before the rpc had 1 link per call, so 1 made sense.
Now we share the link across many calls. You can’t use 1

Comment (serbrech) [reply rpc/rpc.go:46]: It doesn’t really matter what’s the value, unless you reach that kind of rpc call concurrency.

Comment (jhendrixMSFT) [reply rpc/rpc.go:46]: Ah ok yes that makes sense.

Comment (serbrech) [reply rpc/rpc.go:46]: var Receiver = function (session, name, local_handle, opts) {
    this.init(session, name, local_handle, opts, true);
    this.drain = false;
    this.set_credit_window(this.get_option('credit_window', 1000));
    
    in the JS framework

Comment (richardpark-msft) [reply rpc/rpc.go:46]: Yes, rhea (the library we use in JS) uses the same limit.

**Pull request merged**